raft/raft.go

```diff
@@ -726,12 +727,19 @@ func (r *raft) Step(m pb.Message) error {
 			// but it will not receive MsgApp or MsgHeartbeat, so it will not create
 			// disruptive term increases
 			r.send(pb.Message{To: m.From, Type: pb.MsgAppResp})
-		} else {
+			return nil
+		case m.Type == pb.MsgPreVote || m.Type == pb.MsgPreVoteResp:
+			// If we receive a MsgPreVote from a node with a lower term number,
```

Comment:
It isn't, it can be dropped as well. Rejecting it just puts the slower (previously partitioned) node into the candidate state (which would happen anyway when the faster node sends a MsgPreVote message with a higher term). 
As an aside, why do we just drop MsgVote messages with older terms instead of rejecting them directly?